app/web/utils/nominatim.ts

```diff
@@ -42,3 +42,21 @@ export const simplifyPlaceDisplayName = (place: NominatimPlace) => {
   }
   return addressParts.join(", ");
 };
+
+export const filterDuplicatePlaces = (places: NominatimPlace[] = []) => {
+  const deduplicatedPlaces = places.reduce((previousRecord, currentPlace) => {
+    const importance = currentPlace.importance ?? 0;
+    const displayName = simplifyPlaceDisplayName(currentPlace);
+
+    return previousRecord[displayName]?.importance >= importance
+      ? previousRecord
+      : { ...previousRecord, [displayName]: currentPlace };
+  }, {} as Record<string, NominatimPlace>);
+
+  const filtered = places.filter((place) => {
+    const displayName = simplifyPlaceDisplayName(place);
+    return deduplicatedPlaces[displayName].place_id === place.place_id;
+  });
```

Comment:
Could this bit not be done as part of the `reduce` too? Seems unnecessary to loop through `places` again when it's already doing some work there.

Reply:
I was thinking this was needed to preserve ordering but actually `Object.values(deduplicatedPlaces)` does that too. will replace it with that